infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/NodeSelectorProvisioner.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2012-2020 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.provision;
+
+import static java.util.Collections.emptyMap;
+
+import com.google.common.base.Splitter;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.inject.Named;
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.commons.annotation.Nullable;
+import org.eclipse.che.workspace.infrastructure.kubernetes.environment.KubernetesEnvironment;
+
+/** Provisions node selector into workspace pod spec. */
+public class NodeSelectorProvisioner implements ConfigurationProvisioner {
+
+  private final Map<String, String> nodeSelectorAttributes;
+
+  @Inject
+  public NodeSelectorProvisioner(
+      @Nullable @Named("che.workspace.pod.node_selector") String nodeSelectorProperty) {
+    this.nodeSelectorAttributes =
+        nodeSelectorProperty != null
+            ? Splitter.on(",").withKeyValueSeparator("=").split(nodeSelectorProperty)
```

Comment:
Can this throw `IllegalArgumentException`, if e.g. the admin misses a comma in the property string?

Reply:
I didn't found such functionality in Splitter. Does it worth rewriting the whole logic to do that?
or we can add condition below if `nodeSelectorProperty.size >0 and nodeSelectorAttributes.size == 0 then throw IllegalArgumentException("invalid format of nodeSelectorProperty") .
wdyt?

Reply:
I was wrong
```
MapSplitter.
     * @throws IllegalArgumentException if the specified sequence does not split into valid map
     *     entries, or if there are duplicate keys
```